### ISSUE TITLE
Edited work with no reasonably valid author list cannot be cited

### ISSUE THREAD

**Issue by acli** [citation-file-format/citation-file-format#334]
In the current schema.json, in the `references` field, a valid reference entry must have an `authors` field. However, when citing edited volumes (for example) in other styles (e.g., APA or Chicago), author list is usually not given; instead, the editor list is given.

I believe requiring `authors` (instead of, for example, “any of authors, editors”) is an error. Other than not aligning with other styles, it is quite possible that we encounter a reference where we know the editor(s) but not the authors. For example, a library of sound clips maintained by a known individual would be correctly cited by specifying the maintainer as an editor, but if the library in question takes contributions from the public, specifying the maintainer as author would not be factually correct even if the maintainer is an author.

**Comment by sdruskat** [comment]
I think you're right, @acli, thanks for flagging this!
I'll mark this for the next feature release (1.3.0). Requiring one of `editors, authors` doesn't break anything that was possible in the past.

**Comment by AdityaSavara** [comment]
I'm not sure authors should be required, as noted in a comment on a related issue. https://github.com/citation-file-format/citation-file-format/issues/329#issuecomment-921218499

**Comment by acli** [comment]
Re @AdityaSavara’s comment, my view is `authors` clearly shouldn’t be required. In fact I’d say even “anyOf authors, editors” is wrong because both APA and Chicago can deal with citations that have no author, no editor.

**Comment by AdityaSavara** [comment]
I am not trying to "convince" anyone, I just want to point out for comparison that the current CiteSoft standard does not require names. It requires providing any unique ID and a Title (that's the minimum entry required).  The concept of the unique ID is explained on another citation-file-format issues thread: https://github.com/citation-file-format/citation-file-format/issues/344

If the unique-id field is added, citation-file-format could switch to the same approach as CiteSoft: that any unique-id and a title are sufficient. Actually, one could argue that even the title should be optional once a unique-id is provided.

This is almost like a philosophical question, as to what the purpose of CiteSoft is. Should it simply provide allowable files, or should it force having some subset of of fields?  

I would suggest that in the ideal world, there should be three classifications of fields:
**Required:** Maybe nothing? 
**Recommended:** Dependent on MediaType, like books would print a warning that it's recommended to have an author or an editor.  And maybe MediaType should always be a recommended field.
**Allowed:* Maybe dependent on media type, maybe not.

I think that CFF should focus on allowed fields, for now, and then have a roadmap to add warnings for recommended fields in the future.

**Comment by sdruskat** [comment]
> my view is `authors` clearly shouldn’t be required. In fact I’d say even “anyOf authors, editors” is wrong because both APA and Chicago can deal with citations that have no author, no editor.

It is important to note that the latter point refers to citation styles, i.e., the way citation metadata is represented in references. Regardless of how you style a reference, the referenced work will still have either authors or editors. And so, one of these should be required.

The case where you don't know the names of the authors/editors is covered in the documentation [here](https://github.com/citation-file-format/citation-file-format/blob/1.2.0/schema-guide.md#how-to-deal-with-unknown-individual-authors). In short, you can have a single authors or editors entity with a suitable name, e.g., `name: anonymous` or `name: unknown`.

In summary, we agree that it should be possible to have references which don't have authors but do have editors and vice versa, and therefore making the required keys for references `anyOf: [authors, editors]` is how we are going to take this forward.

**Comment by sdruskat** [comment]
Closing as resolved via merging #524.
